core/src/main/java/org/elasticsearch/common/xcontent/XContentType.java

```diff
@@ -143,12 +143,16 @@ public int index() {
         return index;
     }
 
-    public abstract String restContentType();
+    public String mediaType() {
+        return mediaTypeName();
+    }
 
     public abstract String shortName();
 
     public abstract XContent xContent();
 
+    public abstract String mediaTypeName();
```

Comment:
"Media type name" refers to just the top-level media type (i.e., `application|text|multipart|message|image|video|audio`) so we will need a different name here if we do keep this method (see below).

But, I don't think that both `mediaType` and `mediaTypeName` (as they are currently named) are needed. We can get by with just one and then just strip off the parameter (the `;` and everything after) in both the value from the `Accept` header field and in the media type being compared against. In this case, the method name `mediaType` is fine.